### PR TITLE
Network: BGP peer holdtime option configuration

### PR DESCRIPTION
**Pull request by ivanovuri**
Hi,
BGP peer holdtime option configuration with proper documentation
Mine need to have ability to set peer's holdtime on lxd side.
I've already created additional optional configuration node and change complementary code with documentation.
This leads to adding new peer configuration `lxc network set lxdbr0 bgp.peers.foo.holdtime=1`
Сonsider including this option please.

### CODE REVIEW

File: doc/reference/network_bridge.md
Comment (tomponline) [inline doc/reference/network_bridge.md:57]: We should indicate in the docs that this value is in seconds.

**Comment by tomponline** [comment]
jenkins: test this please

File: doc/howto/network_bgp.md
Comment (ru-fu) [inline doc/howto/network_bgp.md:71]:
```suggestion
- `bgp.peers.<name>.holdtime` - an optional hold time for the peer session (in seconds; default is 3 minutes)
```

File: doc/reference/network_bridge.md
Comment (ru-fu) [reply doc/reference/network_bridge.md:57]:
```suggestion
`bgp.peers.NAME.holdtime`            | string    | BGP server            | `180`                     | Peer session holdtime (in seconds; optional)
```

Comment (ru-fu) [reply doc/reference/network_bridge.md:57]: If it is in seconds, why is it a string? Shouldn't it be an integer then - or if it's a string, it should accept units?

File: doc/reference/network_physical.md
Comment (ru-fu) [inline doc/reference/network_physical.md:39]: Same as above.

File: lxd/network/driver_common.go
Comment (ivanovuri) [reply lxd/network/driver_common.go:555]: Because holdtime is three times bigger than keepalive timer I simply restrict it to make minimum 3sec keepalive timer.
Do you think is better to have minimum value of 0(zero) like in bgp 
 specifications?

File: doc/reference/network_bridge.md
Comment (ivanovuri) [reply doc/reference/network_bridge.md:57]: Ops yes, sorry my fault, I'll change docs.

File: doc/reference/network_physical.md
Comment (ivanovuri) [reply doc/reference/network_physical.md:39]: Can you help me resolve spellcheck test. Should I simply add holdtimer word to .wordlist.txt file?

File: lxd/network/driver_common.go
Comment (tomponline) [reply lxd/network/driver_common.go:555]: Any preference @stgraber ?

File: doc/reference/network_physical.md
Comment (ru-fu) [reply doc/reference/network_physical.md:39]: No, it should be "hold time" (two words) in the text, as in the other occurrence. The parameter itself is fine since it's marked as code.

Comment (ivanovuri) [reply doc/reference/network_physical.md]: Didn't wait for your reply for an answer and corrected, sorry.
But already returned the file to its previous correct state.
I will be more patient in the future.

Comment (ru-fu) [reply doc/reference/network_physical.md]: Thanks, this should work now.

**Review by ru-fu — APPROVED** [review]
Docs look good. :)

File: lxd/bgp/server.go
Comment (tomponline) [inline lxd/bgp/server.go:440]: This sentence doesn't read well, can you clarify what you mean here please. Thanks
Also comments should end with a full stop.

File: lxd/network/driver_common.go
Comment (tomponline) [reply lxd/network/driver_common.go:555]: What do you mean about needing to add a keep alive option as well?

File: doc/howto/network_bgp.md
Comment (tomponline) [inline doc/howto/network_bgp.md:71]: This is a bit confusing in my opinion. Elsewhere we say the default is 180s and here we say the default is 3 minutes.
I realize they are the same, but it still feels a bit odd to be mixing units like that.
Also if we ever change the default then it'll likely be missed.
So perhaps just remove the default statement here and leave that to the reference table itself.

File: lxd/bgp/server.go
Comment (tomponline) [reply lxd/bgp/server.go:440]: I think we can probably just lose that comment as the default 180s bit isn't handled here anyway.

File: lxd/network/driver_common.go
Comment (tomponline) [inline lxd/network/driver_common.go:761]: full stop "." at end of sentences please.

File: lxd/bgp/server.go
Comment (ivanovuri) [reply lxd/bgp/server.go]: full stop added thanks.
Good or bad but most most of network engineers this of cisco implementation as reference. Here I just try to clarify why i choose this value. As I know Mikrotik is also using 3minutes, but Juniper prefer 90 sec. interval.
I've changes it a little bit, what do you think.

File: doc/howto/network_bgp.md
Comment (ivanovuri) [reply doc/howto/network_bgp.md]: Agreed, this notification is simplified with all comments too.

File: lxd/network/driver_common.go
Comment (ivanovuri) [reply lxd/network/driver_common.go]: Done.
What do you think about this part:
peerHoldTime = "180"
Maybe we need to create something like this
`const DEFAULT_BGP_HOLDTIME = "180"`
and modify
`peerHoldTime = DEFAULT_BGP_HOLDTIME `
We can do such thing with all other variables like `MultihopTtl: 255` in the future.

File: lxd/bgp/server.go
Comment (tomponline) [reply lxd/bgp/server.go]: The default value seems fine, as you say its used in the industry, it was rather the location of the comment where the default wasn't referenced that concerned me. Its fine now thanks.

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
I would like @stgraber to cast his eye over this before its merged.

File: lxd/network/driver_common.go
Comment (tomponline) [reply lxd/network/driver_common.go]: Thanks, no need for constants right now as the default is only defined once in code.

File: lxd/network/driver_common.go
Comment (ivanovuri) [reply lxd/network/driver_common.go:555]: In general, network devices allow to configure both BGP timers and I start to implement ability to configure both option.
But at the end of the day I realized that it complicates configuration little bit.
To make things simple I've added low range of 9 seconds (keepalive is three times lower).
In case we set low bound to zero, maybe (I just consult with you) we should also add keepalive option, but I now thing lets stick one lonely holdtime parameter.

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**
